### PR TITLE
React app search history

### PR DESCRIPTION
**Pull request by ajs476**

### CODE REVIEW

File: posters-project/src/App.js [hunk 1]
```diff
@@ -50,16 +57,21 @@ class App extends React.Component {
   }
 
   handleSearch() {
-    const {movies, inputValue, statusMessage} = this.state;
+    const {movies, inputValue, statusMessage, searchHistory} = this.state;
+    if (searchHistory.length >= 10) {
```
Comment (lucas-aragno) [inline posters-project/src/App.js]: move this 10 to a constant

**Review by lucas-aragno — CHANGES_REQUESTED** [review]

File: posters-project/src/App.js (hunk 1 above)
Comment (ajs476) [reply posters-project/src/App.js]: done. created maxSearchLength and set that to 10 instead.

File: posters-project/src/App.js [hunk 2]
```diff
@@ -2,9 +2,11 @@ import React from 'react';
 import PosterList from './components/PosterList.js';
 import SearchBar from './components/SearchBar.js';
 import Poster from './components/Poster.js';
+import SearchHistory from './components/SearchHistory';
 import posterData from './data/poster-data.json';
 const qhttp = require('q-io/http');
 
+const maxSearchHistory = 10;
```
Comment (lucas-aragno) [inline posters-project/src/App.js]: use all caps for constants

File: posters-project/src/App.js [hunk 3]
```diff
@@ -50,16 +58,21 @@ class App extends React.Component {
   }
 
   handleSearch() {
-    const {movies, inputValue, statusMessage} = this.state;
+    const {movies, inputValue, statusMessage, searchHistory} = this.state;
+    if (searchHistory.length >= maxSearchHistory) {
```
Comment (lucas-aragno) [inline posters-project/src/App.js]: move this to it's own method

**Review by lucas-aragno — CHANGES_REQUESTED** [review]

File: posters-project/src/App.js (hunk 3 above)
Comment (ajs476) [reply posters-project/src/App.js]: moved to updateHistory()

File: posters-project/src/App.js (hunk 2 above)
Comment (ajs476) [reply posters-project/src/App.js]: changed to MAX_SEARCH_HISTORY

File: posters-project/src/App.js [hunk 4]
```diff
@@ -13,9 +15,22 @@ class App extends React.Component {
...
     this.handleSearch = this.handleSearch.bind(this);
+    this.clearHistory = this.clearHistory.bind(this);
+  }
+
+  clearHistory() {
+    this.setState({searchHistory: []});
+  }
+
+  updateHistory(inputValue, searchHistory) {
+    if (searchHistory.length >= MAX_SEARCH_HISTORY) {
```
Comment (lucas-aragno) [inline posters-project/src/App.js]: move this to a different method (the searchHistory pop logic)

**Review by lucas-aragno — CHANGES_REQUESTED** [review]

File: posters-project/src/App.js (hunk 4 above)
Comment (ajs476) [reply posters-project/src/App.js]: Ah. Done.

**Review by lucas-aragno — APPROVED** [review]

**Pull request merged**